### PR TITLE
[GIT PULL] Fixed leaked ring for send_recv test

### PR DESCRIPTION
**Pull request by itaysnir**
Small fix for leaked ring within send_recv.c test.

Added a call for io_uring_queue_exit() in the cleanup. 

----
## git request-pull output:
```
The following changes since commit 9961f57f1fe58a343b8050fa504a0dd1d4fd6425:

  liburing.spec: bump version to 2.4 (2022-11-01 19:52:25 -0600)

are available in the Git repository at:

  https://github.com/itaysnir/liburing master

for you to fetch changes up to fa366315ab499c05e6cc5f082b591a23098d721f:

  yet another indentation (2022-11-04 04:59:19 -0400)

----------------------------------------------------------------
itaysnir (4):
      fixed ring leak for do_send
      changed indentation
      added space
      yet another indentation

 test/send_recv.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)
```
----
<details>
<summary>Click to show/hide pull request guidelines</summary>

## Pull Request Guidelines
1. To make everyone easily filter pull request from the email
notification, use `[GIT PULL]` as a prefix in your PR title.
```
[GIT PULL] Your Pull Request Title
```
2. Follow the commit message format rules below.
3. Follow the Linux kernel coding style (see: https://github.com/torvalds/linux/blob/master/Documentation/process/coding-style.rst).

### Commit message format rules:
1. The first line is title (don't be more than 72 chars if possible).
2. Then an empty line.
3. Then a description (may be omitted for truly trivial changes).
4. Then an empty line again (if it has a description).
5. Then a `Signed-off-by` tag with your real name and email. For example:
```
Signed-off-by: Foo Bar <foo.bar@gmail.com>
```

The description should be word-wrapped at 72 chars. Some things should
not be word-wrapped. They may be some kind of quoted text - long
compiler error messages, oops reports, Link, etc. (things that have a
certain specific format).

Note that all of this goes in the commit message, not in the pull
request text. The pull request text should introduce what this pull
request does, and each commit message should explain the rationale for
why that particular change was made. The git tree is canonical source
of truth, not github.

Each patch should do one thing, and one thing only. If you find yourself
writing an explanation for why a patch is fixing multiple issues, that's
a good indication that the change should be split into separate patches.

If the commit is a fix for an issue, add a `Fixes` tag with the issue
URL.

Don't use GitHub anonymous email like this as the commit author:
```
123456789+username@users.noreply.github.com
```

Use a real email address!

### Commit message example:
```
src/queue: don't flush SQ ring for new wait interface

If we have IORING_FEAT_EXT_ARG, then timeouts are done through the
syscall instead of by posting an internal timeout. This was done
to be both more efficient, but also to enable multi-threaded use
the wait side. If we touch the SQ state by flushing it, that isn't
safe without synchronization.

Fixes: https://github.com/axboe/liburing/issues/402
Signed-off-by: Jens Axboe <axboe@kernel.dk>
```

</details>

----
## By submitting this pull request, I acknowledge that:
1. I have followed the above pull request guidelines.
2. I have the rights to submit this work under the same license.
3. I agree to a Developer Certificate of Origin (see https://developercertificate.org for more information).

### CODE REVIEW

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]
Please follow the pull request guidelines. Especially the commit message part.

**Comment by itaysnir** [comment]
> Please follow the pull request guidelines. Especially the commit message part.

Done

File: test/send_recv.c [hunk 1]
```diff
@@ -214,18 +214,20 @@ static int do_send(void)
 	ret = io_uring_wait_cqe(&ring, &cqe);
 	if (cqe->res == -EINVAL) {
 		fprintf(stdout, "send not supported, skipping\n");
-		close(sockfd);
-		return 0;
-	}
+	    goto err;
+    }
```
Comment (ammarfaizi2) [inline test/send_recv.c]: Please use tabs as the indentation. Don't use spaces.

File: test/send_recv.c [hunk 2]
```diff
@@ -214,18 +214,20 @@ static int do_send(void)
...
 		fprintf(stderr, "failed cqe: %d\n", cqe->res);
 		goto err;
 	}
 
 	close(sockfd);
 	return 0;
+
 err:
-	close(sockfd);
+    close(sockfd);
```
Comment (ammarfaizi2) [inline test/send_recv.c]: Please use tabs as the indentation. Don't use spaces.

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]
You use:
```
From: itaysnir <itaysnir3687@gmail.com>
```
as the author of the commit. But the `Signed-off-by` is:
```
Signed-off-by: Itay Snir <itaysnir@campus.technion.ac.il>
```
Please fix that, use a consistent name and email. You can set it globally like this:

```
git config --global user.name "Itay Snir";
git config --global user.email "itaysnir@campus.technion.ac.il";
```

**Comment by ammarfaizi2** [comment]
Also, just squash the coding style fix commits into the previous one.

You just introduced a bad indentation and then you fixed it in the
next commit. Don't do that.

There is no need to add extra commit for fixing your non-upstream
mistakes. Just amend the commit if you did something wrong.

**Comment by axboe** [comment]
Yes, please squash it into a single commit, we don't want broken intermediates.

File: test/send_recv.c (hunk 1 above)
Comment (itaysnir) [reply test/send_recv.c]: Done

File: test/send_recv.c (hunk 2 above)
Comment (itaysnir) [reply test/send_recv.c]: Done

**Comment by itaysnir** [comment]
> Also, just squash the coding style fix commits into the previous one.
> 
> You just introduced a bad indentation and then you fixed it in the next commit. Don't do that.
> 
> There is no need to add extra commit for fixing your non-upstream mistakes. Just amend the commit if you did something wrong.

Done :)

**Comment by ammarfaizi2** [comment]
> Done :)

One more missing, please fix the author of the commit. The author must
match with the `Signed-off-by` tag.

If you check your commit with `git log --pretty=fuller`, you will see:
```
Author:     itaysnir <itaysnir3687@gmail.com>
AuthorDate: Fri Nov 4 04:41:49 2022 -0400
Commit:     Itay Snir <itaysnir@campus.technion.ac.il>
CommitDate: Sat Nov 5 22:22:21 2022 -0400
```

But your `Signed-off-by` is:
```
Signed-off-by: Itay Snir <itaysnir@campus.technion.ac.il>
```

See the Author part? If you amend a commit, it doesn't change the author.
It only changes the committer.

You need to do the following:
```
git commit --amend --author "Itay Snir <itaysnir@campus.technion.ac.il>" --no-edit
```

then force push.

**Comment by itaysnir** [comment]
> > Done :)
> 
> One more missing, please fix the author of the commit. The author must match with the `Signed-off-by` tag.
> 
> If you check your commit with `git log --pretty=fuller`, you will see:
> 
> ```
> Author:     itaysnir <itaysnir3687@gmail.com>
> AuthorDate: Fri Nov 4 04:41:49 2022 -0400
> Commit:     Itay Snir <itaysnir@campus.technion.ac.il>
> CommitDate: Sat Nov 5 22:22:21 2022 -0400
> ```
> 
> But your `Signed-off-by` is:
> 
> ```
> Signed-off-by: Itay Snir <itaysnir@campus.technion.ac.il>
> ```
> 
> See the Author part? If you amend a commit, it doesn't change the author. It only changes the committer.
> 
> You need to do the following:
> 
> ```
> git commit --amend --author "Itay Snir <itaysnir@campus.technion.ac.il>" --no-edit
> ```
> 
> then force push.

Thank you for your patience, lol.
Done

File: test/send_recv.c [hunk 3]
```diff
@@ -214,18 +214,21 @@ static int do_send(void)
 	ret = io_uring_wait_cqe(&ring, &cqe);
 	if (cqe->res == -EINVAL) {
 		fprintf(stdout, "send not supported, skipping\n");
-		close(sockfd);
-		return 0;
-	}
+		goto err;
+        }
```
Comment (ammarfaizi2) [inline test/send_recv.c:218]: I missed this. One more here, the indentation should use tabs, not spaces.

**Comment by ammarfaizi2** [comment]
> Thank you for your patience, lol.

It's fine. It's normal for a starter like you. I have found many other new
contributors (+outside liburing) worse than you. They don't even know how to
amend a commit. When I asked to amend or squash their commits, they went away
and didn't come back.

So for me, it's much easier teaching a new contributor who knows how to address
reviews like you. You can easily follow what I said.

File: test/send_recv.c (hunk 3 above)
Comment (itaysnir) [reply test/send_recv.c]: Done

**Review by ammarfaizi2 — APPROVED** [review]
Looks good.

**Pull request merged**